src/main/java/apoc/merge/Merge.java

```diff
@@ -53,18 +76,47 @@ private String wrapInBacktics(String s) {
         return Stream.of(new RelationshipResult(rel));
     }
 
+    @Procedure(value = "apoc.merge.relationship.eager", mode = Mode.WRITE, eager = true)
+    @Description("apoc.merge.relationship.eager(startNode, relType,  identProps:{key:value, ...}, config:{onCreateProps:{key:value, ...}, onMatchProps:{key:value, ...}}, endNode) - merge relationship with dynamic type, with support for setting properties ON CREATE or ON MATCH")
+    public Stream<RelationshipResult> relationshipEager(@Name("startNode") Node startNode, @Name("relationshipType") String relType,
+                                                   @Name("identProps") Map<String, Object> identProps, @Name("config") Map<String, Object> config, @Name("endNode") Node endNode) {
+        validateKeys(config.keySet(), Lists.newArrayList("onCreateProps", "onMatchProps"));
```

Comment:
shouldn't this just delegate to the non-eager version?

Reply:
The non-eager version can only merge a single result (see `Iterators.single()` usage) and doesn't handle ON MATCH SET.